### PR TITLE
DL: Remove reshaping for input/validation dataset

### PR DESCRIPTION
**Pull request by jingyimei**
This PR does the following:
1. Add a simple validation check to fit() that input shape = model architecture expected shape.
2. Do not re-shape image inputs both for training data and validation data
3. Add numeric array check to dependent var for validation table
4. Rename helper class/file to validator
5. Use FORMAT class for getting model_arch col names(previously it was hardcoded, which caused discrepancy between minibatch_preprocessor_dl() and madlib_keras_fit()).
6. Update dev check and unit test(also move all unit tests to madlib_keras.py_in)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/817/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/818/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/819/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/820/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/821/

File: src/ports/postgres/modules/deep_learning/madlib_keras.py_in [hunk 1]
```diff
@@ -487,31 +459,23 @@ def evaluate1(schema_madlib, model_table, test_table, id_col, model_arch_table,
...
-    plpy.info('model in str is {}'.format(str(model)))
 
-    model_shapes = []
-    for weight_arr in model.get_weights():
-        model_shapes.append(weight_arr.shape)
-    _, model_weights = KerasWeightsSerializer.deserialize_weights(
+    loss_acc = get_loss_acc_from_keras_eval(schema_madlib, test_table, dependent_varname,
+                                            independent_varname, compile_params, model_arch,
+                                            model_data, False)
+    #TODO remove these infos after adding create table command
```
[the remark this answers is not in the document]
Comment (kaknikhil) [reply src/ports/postgres/modules/deep_learning/madlib_keras.py_in]: we added this as a reminder to remove them in a future PR when we work on the `evaluate` function.

File: src/ports/postgres/modules/deep_learning/test/madlib_keras.sql_in [hunk 2]
```diff
@@ -17,20 +17,31 @@
...
+ *//* ---------------------------------------------------------------------*/
 
 drop table if exists cifar_10_sample;
 create table cifar_10_sample(
 		id INTEGER,
-		x  REAL[],
-        y SMALLINT,
-        imgpath TEXT
-        );
+    y SMALLINT,
```
[the remark this answers is not in the document]
Comment (kaknikhil) [reply src/ports/postgres/modules/deep_learning/test/madlib_keras.sql_in]: will take care of it

File: src/ports/postgres/modules/deep_learning/madlib_keras.py_in [hunk 3]
```diff
@@ -487,31 +459,23 @@ def evaluate1(schema_madlib, model_table, test_table, id_col, model_arch_table,
...
-    os.environ["CUDA_VISIBLE_DEVICES"] = '-1'
-
-    model = model_from_json(model_arch)
-    plpy.info('model in str is {}'.format(str(model)))
 
-    model_shapes = []
-    for weight_arr in model.get_weights():
-        model_shapes.append(weight_arr.shape)
-    _, model_weights = KerasWeightsSerializer.deserialize_weights(
+    loss_acc = get_loss_acc_from_keras_eval(schema_madlib, test_table, dependent_varname,
```
[the remark this answers is not in the document]
Comment (kaknikhil) [reply src/ports/postgres/modules/deep_learning/madlib_keras.py_in]: We will create a table out of the loss and acc values. This will be done in a future PR.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/823/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/824/
